.github/workflows/build.yml

```diff
@@ -15,12 +15,16 @@ jobs:
     name: Compile the app 
     runs-on: ubuntu-18.04
 
-    steps: 
+    steps:
+    - name: Get Docker Hub username
+      id: get-docker-hub-username
+      run: echo '::set-output name=dockerhub_username::${{ secrets.DOCKERHUB_USERNAME }}'
     - name: Login to Docker Hub
       uses: docker/login-action@v1
       with:
         username: ${{ secrets.DOCKERHUB_USERNAME }}
         password: ${{ secrets.DOCKERHUB_TOKEN }}
+      if: steps.get-docker-hub-username.outputs.dockerhub_username
```

Comment:
Can't we just do `if: secrets.DOCKERHUB_USERNAME` ?

Reply:
Ugh... apparently we can't ... https://github.community/t/if-expression-with-context-variable/16558